packages/sakuli-rollup-hooks/src/ts-support/get-sakuli-version.function.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { getSakuliVersion } from "./get-sakuli-version.function"
+import { readJson } from "../read-json.function"
+import { join } from "path"
+const execa: jest.Mock = require("execa");
+
+jest.mock('../read-json.function.ts', () => ({
+    readJson: jest.fn()
+}))
+
+jest.mock('execa', () => jest.fn());
+
+describe('getSakuliVersion', () => {
+    it('should use version from dependencies in package.json', async () => {
+        (<jest.Mock>readJson).mockResolvedValueOnce({
+            dependencies: { '@sakuli/cli': '2.2.0-file' },
+            devDependencies: { '@sakuli/cli': '2.2.0-dev' }
+        });
+        const version = await getSakuliVersion('some/path');
+        expect(version).toBe('2.2.0-file');
+        expect(readJson).toHaveBeenCalledWith(join('some/path', 'package.json'));
+    })
+
+    it('should use version from devDependencies in package.json when not in dependencies', async () => {
+        (<jest.Mock>readJson).mockResolvedValueOnce({
+            devDependencies: { '@sakuli/cli': '2.2.0-dev' }
+        });
+        const version = await getSakuliVersion('some/path');
+        expect(version).toBe('2.2.0-dev');
+        expect(readJson).toHaveBeenCalledWith(join('some/path', 'package.json'));
+    });
+
+    it('should fallback to output of npx command when package.json cannot be read', async () => {
+        (<jest.Mock>readJson).mockRejectedValue(null);
+        (<jest.Mock>execa).mockResolvedValue({ stdout: '2.2.0-cli' });
+        const version = await getSakuliVersion('some/path');
+        expect(version).toBe('2.2.0-cli');
+        expect(execa).toHaveBeenCalledWith('npx', ['sakuli', '--version'], expect.objectContaining({ cwd: 'some/path' }));
+    });
+
+    it('should fallback to output of npx command when sakuli is not in the dependencies', async () => {
+        (<jest.Mock>readJson).mockRejectedValue({
+            dependencies: {},
+            devDependencies: {},
+        });
+        (<jest.Mock>execa).mockResolvedValue({ stdout: '2.2.0-cli' });
+        const version = await getSakuliVersion('some/path');
+        expect(version).toBe('2.2.0-cli');
+        expect(execa).toHaveBeenCalledWith('npx', ['sakuli', '--version'], expect.objectContaining({ cwd: 'some/path' }));
+    });
+
+    it('should fallback to @latest when all other mechanisms not working', async () => {
+        (<jest.Mock>readJson).mockRejectedValue(null);
+        (<jest.Mock>execa).mockRejectedValue(null);
+        const version = await getSakuliVersion('some/path');
+        expect(version).toBe('latest');
+    });
```

Comment:
Im just curious. Is there any scenario that this might happen? I mean you need `npx sakuli enable-typescript` at minimum.

Reply:
It should just ensure that in every possible case a valid version is returned.
Especially a run command could behave differently on various platforms.